lib/net/pool.js

```diff
@@ -2012,6 +2130,157 @@ class Pool extends EventEmitter {
     peer.sendCFCheckpt(packet.filterType, packet.stopHash, filterHeaders);
   }
 
+  /**
+   * Handle peer `CFCheckpt` packet.
+   * @method
+   * @private
+   * @param {Peer} peer
+   * @param {CFCheckptPacket} packet
+   */
+
+  async handleCFCheckpt(peer, packet) {
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+    }
+  }
+
+  /**
+   * Handle peer `CFHeaders` packet.
+   * @method
+   * @private
+   * @param {Peer} peer - Sender.
+   * @param {CFHeadersPacket} packet - Packet to handle.
+   * @returns {void}
+   */
+  async handleCFHeaders(peer, packet) {
+    this.logger.info('Received CFHeaders packet from %s', peer.hostname());
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const filterType = packet.filterType;
+
+    if (filterType !== this.requestedFilterType) {
+      this.logger.warning('Received CFHeaders packet with wrong filterType');
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const stopHash = packet.stopHash;
+    if (!stopHash.equals(this.requestedStopHash)) {
+      this.logger.warning('Received CFHeaders packet with wrong stopHash');
+      peer.ban();
+      return;
+    }
+    let previousFilterHeader = packet.previousFilterHeader;
+    const filterHashes = packet.filterHashes;
+    let blockHeight = await this.chain.getHeight(stopHash)
+      - filterHashes.length + 1;
+    const stopHeight = await this.chain.getHeight(stopHash);
+    for (const filterHash of filterHashes) {
+      if (blockHeight > stopHeight) {
+        peer.ban();
+        return;
+      }
+      const basicFilter = new BasicFilter();
+      basicFilter._hash = filterHash;
+      const filterHeader = basicFilter.header(previousFilterHeader);
+      const lastFilterHeader = this.cfHeaderChain.tail;
+      const blockHash = await this.chain.getHash(blockHeight);
+      const cfHeaderEntry = new CFHeaderEntry(blockHash,
+        filterHeader, lastFilterHeader.height + 1);
+      this.cfHeaderChain.push(cfHeaderEntry);
+      previousFilterHeader = filterHeader;
+      blockHeight++;
+      this.logger.info('CFHeader height: %d', this.cfHeaderChain.tail.height);
```

Comment:
move this log out of the loop else this will lead to spamming

Reply:
resolved